### PR TITLE
feat: Allow BCC in session notify API

### PR DESCRIPTION
**Pull request by sachinchauhan2889**
Fixes #https://github.com/fossasia/open-event-frontend/issues/5309

required for https://github.com/fossasia/open-event-frontend/pull/6112

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/sessions.py [hunk 1]
```diff
@@ -373,24 +373,27 @@ def notify_for_session(session, mail_override: Dict[str, str] = None):
     # Email for owner
     if session.event.get_owner():
         owner = session.event.get_owner()
-        send_email_session_state_change(owner.email, session, mail_override)
+        send_email_session_state_change(owner.email, session, mail_override, bcc)
         send_notif_session_state_change(
             owner, session.title, session.state, session.site_link, session.id
         )
+    # if bcc:
+    #     send_email_session_state_change(bcc, session, mail_override)
```
Comment (sachinchauhan2889) [inline app/api/sessions.py:381]: @iamareebjamal sir, can i use this as it is sending mail to bcc.

**Comment by sachinchauhan2889** [comment]
@iamareebjamal sir, i am not able to figure-out why bcc is not working.

File: app/api/sessions.py (hunk 1 above)
Comment (iamareebjamal) [reply app/api/sessions.py:381]: Why? It should send using above code already

File: app/api/sessions.py [hunk 2]
```diff
@@ -373,24 +373,27 @@ def notify_for_session(session, mail_override: Dict[str, str] = None):
...
 
 
 @sessions_blueprint.route('/<int:id>/notify', methods=['POST'])
 @api.has_permission('is_speaker_for_session', methods="POST")
 def notify_session(id):
     session = Session.query.filter_by(deleted_at=None, id=id).first_or_404()
 
-    data, errors = SessionNotifySchema().load(request.json)
+    data, errors= SessionNotifySchema().load(request.json)
+    bcc= request.json.get('bcc')
```
Comment (iamareebjamal) [inline app/api/sessions.py:390]: No, add bcc in the schema and load from there

File: app/api/sessions.py [hunk 3]
```diff
@@ -373,24 +373,27 @@ def notify_for_session(session, mail_override: Dict[str, str] = None):
...
     if errors:
         raise UnprocessableEntityError(
             {'pointer': '/data', 'errors': errors}, 'Data in incorrect format'
         )
 
-    notify_for_session(session, data)
+    notify_for_session(session, data, bcc)
```
Comment (iamareebjamal) [inline app/api/sessions.py:396]: bcc will be in data itself

**Comment by iamareebjamal** [comment]
How are you checking BCC?

File: app/api/helpers/mail.py [hunk 4]
```diff
@@ -185,6 +185,7 @@ def send_email_session_state_change(email, session, mail_override: Dict[str, str
             mail = mail.copy()
             mail['subject'] = mail_override.get('subject') or mail['subject']
             mail['message'] = mail_override.get('message') or mail['message']
+            mail['bcc'] = mail_override.get('bcc') or mail['bcc']
```
Comment (iamareebjamal) [inline app/api/helpers/mail.py:188]: There is no mail['bcc'], this line is unnecessary

Comment (iamareebjamal) [inline app/api/helpers/mail.py:188]:
```suggestion
            mail['bcc'] = mail_override.get('bcc') or ]]
```

File: app/api/sessions.py [hunk 5]
```diff
@@ -384,7 +384,7 @@ def notify_for_session(session, mail_override: Dict[str, str] = None):
 def notify_session(id):
     session = Session.query.filter_by(deleted_at=None, id=id).first_or_404()
 
-    data, errors = SessionNotifySchema().load(request.json)
+    data, errors= SessionNotifySchema().load(request.json)
```
Comment (iamareebjamal) [inline app/api/sessions.py:387]: Why?

**Comment by iamareebjamal** [comment]
You did not run `pre-commit install` in your virtual environment?

**Comment by sachinchauhan2889** [comment]
> You did not run `pre-commit install` in your virtual environment?

i did. is it related to celeraybeat.schedule file.

File: app/api/sessions.py (hunk 5 above)
Comment (sachinchauhan2889) [reply app/api/sessions.py:387]: by mistake. i will revert this.

File: app/api/helpers/mail.py (hunk 4 above)
Comment (sachinchauhan2889) [reply app/api/helpers/mail.py:188]: sir do i need to remove this line or not.

File: app/api/sessions.py (hunk 5 above)
Comment (iamareebjamal) [reply app/api/sessions.py]: This should have been automatically fixed by pre-commit

File: app/api/helpers/mail.py [hunk 6]
```diff
@@ -161,7 +161,7 @@ def send_email_new_session(email, event_name, link):
     )
 
 
-def send_email_session_state_change(email, session, mail_override: Dict[str, str] = None):
+def send_email_session_state_change(email, session, mail_override: Dict[str, str]=None):
```
Comment (iamareebjamal) [inline app/api/helpers/mail.py:164]: Why

File: app/api/helpers/mail.py [hunk 7]
```diff
@@ -185,6 +185,7 @@ def send_email_session_state_change(email, session, mail_override: Dict[str, str
             mail = mail.copy()
             mail['subject'] = mail_override.get('subject') or mail['subject']
             mail['message'] = mail_override.get('message') or mail['message']
+            mail['bcc'] = mail_override.get('bcc') or []
```
Comment (iamareebjamal) [inline app/api/helpers/mail.py:188]:
```suggestion
            mail['bcc'] = mail_override.get('bcc', [])
```

**Comment by iamareebjamal** [comment]
Remove celerybeat-schedule

**Pull request merged**